rules/qc.smk

```diff
@@ -0,0 +1,225 @@
+        """
+
+
+rule rseqc_readdis:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+        "star/rseqc/annotation.bed"
+    output:
+        "star/rseqc/{sample}-{unit}.readdistribution.txt"
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_readdis/{sample}-{unit}.stderr"
+    params:
+        extra = r''
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        read_distribution.py {params.extra} -r {input[1]} -i {input[0]} > {output} 2> {log}
+        """
+
+
+rule rseqc_readdup:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+    output:
+        touch('star/rseqc/{sample}-{unit}.readdup.done')
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_readdup/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_readdup/{sample}-{unit}.stderr"
+    params:
+        extra = r'',
+        prefix = "star/rseqc/{sample}-{unit}.readdup"
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        read_duplication.py {params.extra} -i {input[0]} -o {output} > {log[0]} 2> {log[1]}
+        """
+
+        
+rule rseqc_readgc:
+    input:
+        "star/{sample}-{unit}/Aligned.out.bam",
+    output:
+        touch('star/rseqc/{sample}-{unit}.readgc.done')
+    priority: 1
+    log:
+        "logs/rseqc/rseqc_readgc/{sample}-{unit}.stdout",
+        "logs/rseqc/rseqc_readgc/{sample}-{unit}.stderr"
+    params:
+        extra = r'',
+        prefix = "star/rseqc/{sample}-{unit}.readgc"
+    conda:
+        "../envs/rseqc.yaml"
+    shell:
+        """
+        read_GC.py {params.extra} -i {input[0]} -o {params.prefix} > {log[0]} 2> {log[1]}
+        """
+        
+
+rule multiqc:
+    input:
+        expand("star/{unit.sample}-{unit.unit}/Log.final.out", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.junction_annotation.done", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.junction_saturation.done", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.infer_experiment.txt", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.stats.txt", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.innerdistance.done", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.readdistribution.txt", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.readdup.done", unit=units.itertuples()),
+        expand("star/rseqc/{unit.sample}-{unit.unit}.readgc.done", unit=units.itertuples())
+    output:
+        "qc/multiqc_report.html"
+    log:
+        "logs/multiqc.stdout",
+        "logs/multiqc.stderr"
+    conda:
+        "../envs/multiqc.yaml"
+    params:
+        extra = '',
+        out = "qc",
+        align = "star",
+        rseqc = "star/rseqc",
+        rseqc_anno = "logs/rseqc/rseqc_junction_annotation",
+        rseqc_dis = "logs/rseqc/rseqc_innerdis"
+    shell:
```

Comment:
There is a wrapper for multiqc here: https://snakemake-wrappers.readthedocs.io/en/stable/wrappers/multiqc.html. We can just use that and don't have to care about the conda environment and the exact command.

Reply:
So I need two directories from the logs as input to multiqc...

"logs/rseqc/rseqc_junction_annotation",
"logs/rseqc/rseqc_innerdis"

these are no explicit outputs of any rule but are created because of log files. How should I do this with the multiqc wrapper?  or specify them as explicit outputs of the rules that create the logs?

Reply:
You specify the concrete files needed as input to multiqc (files from the log directive can also be used as input files of other rules). Then, in the multiqc rule, you have something like this:

```
    params:
        rseqc_anno=lambda _, input: os.path.dirname(input.rseqc_anno_log)
```

Reply:
Done